core/templates/cowdata.h

```diff
@@ -220,26 +220,63 @@ class CowData {
 	Error resize(Size p_size);
 
 	_FORCE_INLINE_ void remove_at(Size p_index) {
-		ERR_FAIL_INDEX(p_index, size());
-		T *p = ptrw();
 		Size len = size();
-		for (Size i = p_index; i < len - 1; i++) {
-			p[i] = p[i + 1];
+		ERR_FAIL_INDEX(p_index, len);
+
+		if (len == 1) {
+			// Removing the only element.
+			clear();
+			return;
 		}
 
-		resize(len - 1);
+		T *ptr = ptrw();
+		T *ptr_idx = ptr + p_index;
+		if constexpr (std::is_trivially_destructible_v<T>) {
+			// Move every element one index down.
+			memmove(reinterpret_cast<uint8_t *>(ptr_idx), reinterpret_cast<uint8_t *>(ptr_idx + 1), (len - p_index - 1) * sizeof(T));
+			resize(len - 1);
+		} else {
+			// TODO This could be more efficient with an explicit destructor and unchecked shrink, but they don't exist yet.
```

Comment:
I don't think this is safe, I think we need to do a semantic move here, we can't rely on a memmove as we don't know if say, the elements have pointers to themselves. We have to call ctors and dtors here, *or* use an std::move.

Reply:
That seems reasonable. I'll make this PR a draft until we have a `is_trivially_relocatable`-like type trait. Until then, I will optimize this function by using move semantics in another PR.